### ISSUE TITLE
Spreadsheet: ISOWEEKNUM(x) incorrect

### ISSUE THREAD

**Issue by jonathanwin** [ONLYOFFICE/DocumentServer#1686]
**Do you want to request a *feature* or report a *bug*?**
* Bug

**What is the current behavior?**
* ISOWEEKNUM() is incorrect for years that start on Friday, Saturday, or Sunday, returning "1" for these days, and "2" for the following week (which should be week "1"), etc.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
* Calculate ISOWEEKNUM("1/1/2022"), it will return "1" instead of "52"
* All days in 2021, 2022, 2023, 2027, 2028, etc.  will have an incorrect ISO-8601 week number
* See file: [isoweek_test.xlsx](https://github.com/ONLYOFFICE/DocumentServer/files/8265577/isoweek_test.xlsx)

**What is the expected behavior?**
* ISOWEEKNUM() should return ISO-8601 weeks: it should start counting at "1" starting on the first week whose Thursday is in the new year (weeks start on Monday). Any days preceding this week are numbered as part of the last week of the previous year (52 or 53)
* This seems properly implemented in the function WEEKNUM(x;21)

**Did this work in previous versions of DocumentServer?**
* Unknown

**DocumentServer version:**
* 7.0.0.132

**Operating System:**
* Linux

**Browser version:**
* Firefox 91.6.0esr

**Comment by Rita-Bubnova** [comment]
Hello, @jonathanwin.
Thank you for the report. 
I can confirm - this is bug, I created new issue 56154 in our private issue tracker.

**Comment by DutchPete** [comment]
This has not been fixed in OnlyOffice version 7.2 either, that is 6 months after filing a bug.

**Comment by Rita-Bubnova** [comment]
Sorry I'm just a QA specialist, nothing I can do. You better contact sales@onlyoffice.com.

**Comment by rsnegirev** [comment]
The problem was fixed in this commit https://github.com/ONLYOFFICE/sdkjs/commit/e681ef667b2497cde75e3ad5bbeafdb1628ab5e2

It will be released in the next major version.

**Comment by DutchPete** [comment]
Hopefully it will not be fixed in Linux only, as reported, but on the other platforms too.

For any prospective users looking for an interim solution, I have found a formula that is a brilliant workaround. On this [webpage](https://www.reddit.com/r/excel/comments/3v55jo/weeknum_numbers_are_wrong_for_2016/?rdt=65159) see the comment by LanMarkx.

I checked it: it works on years where Jan. 1st is week 1 and where it is week 53.
